integrations/SD_Lon/sd_fix_departments.py

```diff
@@ -0,0 +1,137 @@
+import os
+import logging
+import requests
+import datetime
+import sd_payloads
+
+from sd_common import sd_lookup
+from os2mo_helpers.mora_helpers import MoraHelper
+
+LOG_LEVEL = logging.DEBUG
+LOG_FILE = 'fix_sd_departments.log'
+
+logger = logging.getLogger("sdFixDepartments")
+
+detail_logging = ('sdCommon', 'sdFixDepartments')
+for name in logging.root.manager.loggerDict:
+    if name in detail_logging:
+        logging.getLogger(name).setLevel(LOG_LEVEL)
+    else:
+        logging.getLogger(name).setLevel(logging.ERROR)
+
+logging.basicConfig(
+    format='%(levelname)s %(asctime)s %(name)s %(message)s',
+    level=LOG_LEVEL,
+    filename=LOG_FILE
+)
+
+MORA_BASE = os.environ.get('MORA_BASE', None)
+
+
+class FixDepartmentsSD(object):
+    def __init__(self, from_date):
+        logger.info('Start program')
+        self.helper = MoraHelper(hostname=MORA_BASE, use_cache=False)
+        self.from_date = from_date
+
+        try:
+            self.org_uuid = self.helper.read_organisation()
+        except requests.exceptions.RequestException as e:
+            logger.error(e)
+            print(e)
+            exit()
+
+        logger.info('Read org_unit types')
+        self.unit_types = self.helper.read_classes_in_facet('org_unit_type')[0]
+
+    def _create_department(self, department, activation_date):
+        params = {
+            'ActivationDate': activation_date,
+            'DeactivationDate': activation_date,
+            'DepartmentIdentifier': department['DepartmentIdentifier'],
+            'ContactInformationIndicator': 'true',
+            'DepartmentNameIndicator': 'true',
+            'PostalAddressIndicator': 'false',
+            'ProductionUnitIndicator': 'false',
+            'UUIDIndicator': 'true',
+            'EmploymentDepartmentIndicator': 'false'
+        }
+        department_info = sd_lookup('GetDepartment20111201', params)
+
+        for unit_type in self.unit_types:
+            if unit_type['user_key'] == department['DepartmentLevelIdentifier']:
+                unit_type_uuid = unit_type['uuid']
+
+        # TODO: MO currently fails when creating future sub-units
+        activation_date = '2019-07-01'  # Temporary!!!!
+
+        payload = sd_payloads.create_org_unit(
+            department=department,
+            org=self.org_uuid,
+            name=department_info['Department']['DepartmentName'],
+            unit_type=unit_type_uuid,
+            from_date=activation_date
+        )
+        logger.debug('Create department payload: {}'.format(payload))
+
+        response = self.helper._mo_post('ou/create', payload)
+        assert response.status_code == 201
```

Comment:
Jeg har læst lidt på assertlektien.
assert betyder: 
```
if __debug__:
    if not expression: raise AssertionError
```
Det er helt ok i normale tilfælde, fordi __debug__ pr. default sættes til True
Hvis nogen imidlertid finder på at køre programmet med -O for optimize, vil __debug__ være falsk.
-
På den måde kan vi rent faktisk få nogle fejl, hvor vi står og river hinanden i håret:
Jamen den SKAL jo være 201 - det står lige der - uden mulighed for at se i loggen hvilke parametre, programmet blev kørt med.
.
Jeg vil lige i dette tilfælde foretrække noget ala:
```
if not code==201:
    raise noget.
```
Eller  i dette specifikke tilfælde:
response.raise_for_status()